### PR TITLE
ml_evaluate() for clustering models

### PR DESCRIPTION
**Pull request by samuelmacedo83**
Hi, in spark 2.3.0 was implemented Silhouette evaluation for `kmeans`, `bisecting kmeans` and `gmm`, see [here](https://spark.apache.org/docs/2.3.0/ml-clustering.html).
I implemented `ml_evaluate()` for these models. I didn't find any evaluation for `LDA`, so I wrote a warning to this. Here is an example you can run:

```
sc <- spark_connect(master = "local")
iris_tbl <- sdf_copy_to(sc, iris, name = "iris_tbl", overwrite = TRUE)

ml_gaussian_mixture(iris_tbl, Species ~ .) %>%
  ml_evaluate(iris_tbl)

ml_kmeans(iris_tbl, Species ~ .) %>%
  ml_evaluate(iris_tbl)

ml_bisecting_kmeans(iris_tbl, Species ~ .) %>%
  ml_evaluate(iris_tbl)
```

### CODE REVIEW

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2456-1/2c26c7c4-7e5b-49a9-ac4e-d65c92126c5b.txt).

File: R/ml_evaluate.R [hunk 1]
```diff
@@ -59,6 +74,30 @@ ml_evaluate.ml_generalized_linear_regression_model <- function(x, dataset) {
...
+      spark_jobj() %>%
+      invoke("transform", spark_dataframe(dataset))
+
+    silhouette <- sc %>%
+      invoke_new("org.apache.spark.ml.evaluation.ClusteringEvaluator") %>%
+      invoke("evaluate", prediction)
+
+    dplyr::tibble(Silhouette = silhouette)
+  } else{
+    warning("Silhouette is only available for spark 2.3.0 or greater.")
```
Comment (yl790) [inline R/ml_evaluate.R:97]:
```suggestion
    stop("Silhouette is only available for spark 2.3.0 or greater.")
```

Throw an error instead? To me it looks like `ml_evaluate` methods should either return a value or an error, so having a 3rd option of returning nothing is a bit unusual.

Comment (samuelmacedo83) [reply R/ml_evaluate.R:97]: I'd agree

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2456-2/06a4c6a5-ed39-48f8-8dec-b908f248847c.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2456-3/83c345cc-d6e4-4ad2-98ea-3d25d4ee96f6.txt).

**Comment by falaki** [comment]
Databricks Connect tests failed. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2456-5/655dfcac-30f3-4c35-b8c5-4b74aae2f18c.txt).

**Comment by falaki** [comment]
Databricks Connect tests failed. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2456-6/80a596ac-158d-4d58-ae8a-0f7833b4964c.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2456-4/5209e56f-10f8-40a0-8877-48afdbb8bb1e.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2456-7/3b9ce770-da80-4c49-b096-5bc2c3fa5bb2.txt).

**Pull request merged**